src/chat/entity_button.rs

```diff
@@ -100,71 +101,105 @@ live_design!(
 );
 
 #[derive(Live, Widget, LiveHook)]
-pub struct AgentButton {
+pub struct EntityButton {
     #[deref]
     view: View,
 
     #[rust]
-    agent: Option<MofaAgent>,
-
-    #[live(false)]
-    create_new_chat: bool,
-
-    #[live(false)]
-    select_agent_on_prompt: bool,
+    entity: Option<ChatEntityId>,
 }
 
-impl Widget for AgentButton {
+impl Widget for EntityButton {
     fn handle_event(&mut self, cx: &mut Cx, event: &Event, scope: &mut Scope) {
         self.view.handle_event(cx, event, scope);
-        self.widget_match_event(cx, event, scope);
     }
 
     fn draw_walk(&mut self, cx: &mut Cx2d, scope: &mut Scope, walk: Walk) -> DrawStep {
         self.view.draw_walk(cx, scope, walk)
     }
 }
 
-impl WidgetMatchEvent for AgentButton {
-    fn handle_actions(&mut self, cx: &mut Cx, actions: &Actions, _scope: &mut Scope) {
-        let Some(agent) = self.agent else { return };
-
+impl EntityButton {
+    pub fn clicked(&self, actions: &Actions) -> bool {
         if let Some(item) = actions.find_widget_action(self.view.widget_uid()) {
             if let ViewAction::FingerDown(fd) = item.cast() {
-                if fd.tap_count == 1 {
-                    if self.create_new_chat {
-                        cx.action(ChatAction::Start(ChatHandler::Agent(agent)));
-                    }
-
-                    if self.select_agent_on_prompt {
-                        cx.action(PromptInputAction::AgentSelected(agent));
-                    }
-                }
+                return fd.tap_count == 1;
             }
         }
+
+        false
     }
-}
 
-impl AgentButton {
-    pub fn set_agent(&mut self, agent: &MofaAgent, show_description: bool) {
+    pub fn get_entity_id(&self) -> Option<&ChatEntityId> {
+        self.entity.as_ref()
+    }
+
+    pub fn set_agent(&mut self, agent: &MofaAgent) {
```

Comment:
Since most usage of this component is still to display agents, this method is preserved but implemented on top of `set_entity`. This avoid propagating `ChatEntityRef` unnecessarily across the code base.